File: app/actions/emails.ts
"use server";

export async function fetchNewEmails(userId: string) {
  const response = await fetch(`${process.env.NEXT_PUBLIC_BACKEND_URL}/emails/fetch/${userId}`, {
    method: "POST",
  });

  if (!response.ok) {
    throw new Error("Failed to fetch new emails");
  }

  return response.json();
}

File: backend/routers/emails.py
from fastapi import APIRouter, HTTPException
from lib.email_processor import process_gmail_messages

router = APIRouter(prefix="/emails", tags=["emails"])


@router.post('/fetch/{user_id}')
def fetch_emails(user_id: str):
    try:
        return process_gmail_messages(user_id)
    except Exception:
        raise HTTPException(status_code=500, detail="Failed to process emails")